### PR TITLE
Functional Tests : WS - Products - CRUD

### PR DESCRIPTION
**Pull request by Progi1984**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | Functional Tests : WS - Products - CRUD
| Type?             | improvement
| Category?         | TE
| BC breaks?        | no
| Deprecations?     | no
| How to test?      | CI is :green_circle: & Nightly is :green_circle: : https://github.com/Progi1984/ga.tests.ui.pr/actions/runs/5683326925
| Fixed ticket?     | N/A
| Related PRs       | N/A
| Sponsor company   | @PrestaShopCorp

Linked issues : 
* #14903 
* #33429
* #33455

### CODE REVIEW

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts [hunk 1]
```diff
@@ -117,6 +123,26 @@ class DescriptionTab extends BOBasePage {
...
+   * @param languageId {string | undefined}
+   */
+  async getValue(page: Page, inputName: string, languageId?: string): Promise<string> {
+    switch (inputName) {
+      case 'description':
+        return this.getTextContent(page, `${this.productDescription}_${languageId}`, false);
+      case 'id_category_default':
+        return page.$eval(this.productDefaultCategory, (node: HTMLSelectElement) => node.value);
+      case 'manufacturer':
+        return page.$eval(this.productManufacturer, (node: HTMLSelectElement) => node.value);
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]: Use `page.evaluate` instead ?

File: tests/UI/pages/BO/catalog/productsV2/add/optionsTab.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    this.productShowPricesRadio = '#product_options_visibility_show_price_1';
+    this.productOnlineOnlyRadio = '#product_options_visibility_online_only_1';
+  }
+
+  /*
+  Methods
+   */
+
+  /**
+   * @param page {Page}
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/optionsTab.ts]: Please add function description

File: tests/UI/pages/BO/catalog/productsV2/add/pricingTab.ts [hunk 3]
```diff
@@ -54,6 +75,31 @@ class PricingTab extends BOBasePage {
     ]);
     await page.locator(`li:has-text('${productData.taxRule}')`).click();
   }
+
+  /**
+   * @param page {Page}
+   * @param inputName {string}
+   */
+  async getValue(page: Page, inputName: string): Promise<string> {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/pricingTab.ts]: same here

File: tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+  /*
+  Methods
+   */
+
+  /**
+   * @param page {Page}
+   * @param inputName {string}
+   * @param languageId {string | undefined}
+   */
+  async getValue(page: Page, inputName: string, languageId?: string): Promise<string> {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts]: same

File: tests/UI/pages/BO/catalog/productsV2/add/shippingTab.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+  /*
+  Methods
+   */
+
+  /**
+   * @param page {Page}
+   * @param inputName {string}
+   * @param languageId {string | undefined}
+   */
+  async getValue(page: Page, inputName: string, languageId?: string): Promise<string> {
```
Comment (boubkerbribri) [inline tests/UI/pages/BO/catalog/productsV2/add/shippingTab.ts]: same

File: tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts (hunk 1 above)
Comment (Progi1984) [reply tests/UI/pages/BO/catalog/productsV2/add/descriptionTab.ts]: Fixed

File: tests/UI/pages/BO/catalog/productsV2/add/optionsTab.ts (hunk 2 above)
Comment (Progi1984) [reply tests/UI/pages/BO/catalog/productsV2/add/optionsTab.ts]: Fixed

File: tests/UI/pages/BO/catalog/productsV2/add/pricingTab.ts (hunk 3 above)
Comment (Progi1984) [reply tests/UI/pages/BO/catalog/productsV2/add/pricingTab.ts]: Fixed

File: tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts (hunk 4 above)
Comment (Progi1984) [reply tests/UI/pages/BO/catalog/productsV2/add/seoTab.ts]: Fixed

File: tests/UI/pages/BO/catalog/productsV2/add/shippingTab.ts (hunk 5 above)
Comment (Progi1984) [reply tests/UI/pages/BO/catalog/productsV2/add/shippingTab.ts]: Fixed

**Review by boubkerbribri — DISMISSED** [review]

**Comment by Progi1984** [comment]
Ping @boherm pour review

**Comment by Progi1984** [comment]
Ping @nesrineabdmouleh

**Review by boubkerbribri — APPROVED** [review]

**Comment by boubkerbribri** [comment]
Restarting tests https://github.com/boubkerbribri/ga.tests.ui.pr/actions/runs/5879163154

**Review by nesrineabdmouleh — APPROVED** [review]

**Comment by kpodemski** [comment]
@Progi1984 feel free to merge if it's QA approved and regarding your expertise, no need to wait for others 👍🏻

**Comment by boubkerbribri** [comment]
> @Progi1984 feel free to merge if it's QA approved and regarding your expertise, no need to wait for others 👍🏻

I restarted the last test and forget about it, didn't see the QA :white_check_mark: . Don't worry about it :wink:

**Comment by Progi1984** [comment]
@kpodemski It's the job of maintainers. I'm a simple contributor.

**Pull request merged**